apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx

```diff
@@ -137,7 +139,18 @@ export default function ComplexEditorStateDrawer(): JSX.Element {
       )
 
       // TODO: Mark removed images/files as deleted
-      // const assetsToDelete = modifiedAssets.filter((asset) => !asset.file)
+      const assetsToDelete = modifiedAssets.filter(
+        (asset) => asset.src !== undefined,
+      )
+      await Promise.all(
+        assetsToDelete.map(({ src }) => {
+          if (!src) {
+            return
+          }
+
+          return deleteAsset({ fileKey: src })
+        }),
+      )
```

Comment:
I suspect we can update the trpc procedure to take in an array of assets to delete, instead of having this Promise.all.